presto-native-execution/presto_cpp/main/types/PrestoToVeloxQueryPlan.cpp

```diff
@@ -1027,6 +1038,139 @@ core::LocalPartitionNode::Type toLocalExchangeType(
+  }
+}
+
+std::vector<std::string> fromStringList(
+    const protocol::List<protocol::String>& stringList) {
+  std::vector<std::string> stringVector;
+  stringVector.reserve(stringList.size());
+  for (const auto& string : stringList) {
+    stringVector.push_back(string);
+  }
+  return stringVector;
+}
+
+core::SortOrder toSortOrder(protocol::Order order) {
+  switch (order) {
+    case protocol::Order::ASCENDING:
+      return core::SortOrder(true, true);
+    case protocol::Order::DESCENDING:
+      return core::SortOrder(false, false);
+    default:
+      VELOX_USER_FAIL("Unknown sort order: {}", toJsonString(order));
+  }
+}
+
+std::shared_ptr<HiveSortingColumn> toHiveSortingColumn(
+    const protocol::SortingColumn& sortingColumn) {
+  return std::make_shared<HiveSortingColumn>(
+      sortingColumn.columnName, toSortOrder(sortingColumn.order));
+}
+
+std::vector<std::shared_ptr<const HiveSortingColumn>> toHiveSortingColumns(
+    const protocol::List<protocol::SortingColumn>& sortedBy) {
+  std::vector<std::shared_ptr<const HiveSortingColumn>> sortingColumns;
+  sortingColumns.reserve(sortedBy.size());
+  for (const auto& sortingColumn : sortedBy) {
+    sortingColumns.push_back(toHiveSortingColumn(sortingColumn));
+  }
+  return sortingColumns;
+}
+
+std::shared_ptr<HiveBucketProperty> toHiveBucketProperty(
+    const std::vector<std::shared_ptr<const connector::hive::HiveColumnHandle>>&
+        inputColumns,
+    const std::shared_ptr<protocol::HiveBucketProperty>& bucketProperty) {
+  if (bucketProperty == nullptr) {
+    return nullptr;
+  }
+  VELOX_USER_CHECK(
+      bucketProperty->sortedBy.empty(),
+      "Bucketed sorted table is not supported: {}",
+      toJsonString(*bucketProperty));
+
+  const int32_t bucketCount = bucketProperty->bucketCount;
+  if (bucketCount <= 0) {
+    return nullptr;
+  }
+
+  const std::vector<std::string> bucketedColumns =
+      fromStringList(bucketProperty->bucketedBy);
+  VELOX_USER_CHECK(
+      !bucketedColumns.empty(),
+      "Bucketed columns must be set: {}",
+      toJsonString(*bucketProperty));
+
+  const HiveBucketProperty::Kind kind =
+      toHiveBucketPropertyKind(bucketProperty->bucketFunctionType);
+  std::vector<TypePtr> bucketedTypes;
+  if (kind == HiveBucketProperty::Kind::kHiveCompatible) {
+    VELOX_USER_CHECK_NULL(
+        bucketProperty->types,
+        "Unexpected bucketed types set for hive compatible bucket function: {}",
+        toJsonString(*bucketProperty));
+    bucketedTypes.reserve(bucketedColumns.size());
+    for (const auto& bucketedColumn : bucketedColumns) {
+      TypePtr bucketedType{nullptr};
+      for (const auto& inputColumn : inputColumns) {
+        if (inputColumn->name() != bucketedColumn) {
+          continue;
+        }
+        VELOX_USER_CHECK_NOT_NULL(inputColumn->hiveType());
+        bucketedType = inputColumn->hiveType();
+        break;
+      }
+      VELOX_USER_CHECK_NOT_NULL("Bucketed column {} not found", bucketedColumn);
```

Comment:
Is this correct check? Don't you want to check bucketedType for being null?

Reply:
We have checked that bucketedTypes are null in case hive compatible bucket function type in this branch. So we expect to get bucket type from input columns.

Reply:
@mbasmanova oh, yes, bucketedType is missed in the check. Thanks for pointing out!